src/sentry/workflow_engine/migration_helpers/rule.py

```diff
@@ -205,6 +203,39 @@ def update_dcg(
     return dcg
 
 
+def delete_migrated_issue_alert(rule: Rule):
+    try:
+        alert_rule_workflow = AlertRuleWorkflow.objects.get(rule=rule)
+    except AlertRuleWorkflow.DoesNotExist:
+        logger.exception("AlertRuleWorkflow does not exist", extra={"rule_id": rule.id})
+        return
+
+    workflow: Workflow = alert_rule_workflow.workflow
+
+    try:
+        if_dcg = WorkflowDataConditionGroup.objects.get(workflow=workflow).condition_group
+        DataCondition.objects.filter(condition_group=if_dcg).delete()
+        delete_workflow_actions(if_dcg=if_dcg)
+        if_dcg.delete()
+
+    except WorkflowDataConditionGroup.DoesNotExist:
+        logger.exception(
+            "WorkflowDataConditionGroup does not exist", extra={"workflow_id": workflow.id}
+        )
+
+    if not workflow.when_condition_group:
+        logger.error(
+            "Workflow does not have a when_condition_group", extra={"workflow_id": workflow.id}
+        )
+    else:
+        when_dcg = workflow.when_condition_group
+        DataCondition.objects.filter(condition_group=when_dcg).delete()
```

Comment:
do we need to assert that `when_dcg` exists? it can be null, then we would delete floating `DataCondition` objects without a `condition_group`